plugins/utilities/mood_light.py

```diff
@@ -0,0 +1,248 @@
+            position=(100, 200),
+            h_align="center",
+            v_align="center",
+            maxwidth=400.0,
+            text=str(Ldefault),
+            click_activate=True,
+            selectable=True)
+
+        lower_text_label = ba.textwidget(
+            parent=self._root_widget,
+            size=(200, 100),
+            position=(100, 150),
+            h_align="center",
+            v_align="center",
+            text="Limit darkness")
+
+        self.upper_text = ba.textwidget(
+            parent=self._root_widget,
+            size=(200, 100),
+            scale=2,
+            position=(400, 200),
+            h_align="center",
+            v_align="center",
+            maxwidth=400.0,
+            text=str(Udefault),
+            click_activate=True,
+            selectable=True)
+
+        upper_text_label = ba.textwidget(
+            parent=self._root_widget,
+            size=(200, 100),
+            position=(400, 150),
+            h_align="center",
+            v_align="center",
+            text="Limit brightness")
+
+        self.warn_text = ba.textwidget(
+            parent=self._root_widget,
+            text="",
+            size=(400, 200),
+            position=(150, 300),
+            h_align="center",
+            v_align="center",
+            maxwidth=600)
+
+        self.close_button = ba.buttonwidget(
+            parent=self._root_widget,
+            position=(550, 590),
+            size=(35, 35),
+            icon=ba.gettexture("crossOut"),
+            icon_color=(1, 0.2, 0.2),
+            scale=2,
+            color=(1, 0.2, 0.2),
+            extra_touch_border_scale=5,
+            on_activate_call=self.close,
+            button_type="square")
+
+        save_button = ba.buttonwidget(
+            parent=self._root_widget,
+            position=(520, 470),
+            size=(90, 70),
+            scale=1.5,
+            label="SAVE",
+            button_type="square",
+            on_activate_call=self.save_settings)
+
+        ba.textwidget(edit=self.upper_text, on_activate_call=ba.Call(self.on_text_click, "upper"))
+        ba.textwidget(edit=self.lower_text, on_activate_call=ba.Call(self.on_text_click, "lower"))
+
+    def save_settings(self):
+        ba.app.config["moodlightingSettings"] = (Ldefault, Udefault)
+        Print("settings saved")
+        self.close()
+
+    def close(self):
+        ba.containerwidget(edit=self._root_widget, transition="out_right",)
+
+
+Map._old_init = Map.__init__
+
+
+def new_chat_message(msg: Union[str, ba.Lstr], clients: Sequence[int] = None, sender_override: str = None):
+    old_fcm(msg, clients, sender_override)
+    if msg == 'ml':
+        try:
+            Ldefault, Udefault = ba.app.config.get("moodlightingSettings")
+            SettingWindow()
+            _ba.chatmessage("Mood light settings opened")
+        except Exception as err:
+            Print(err)
+
+
+old_fcm = _ba.chatmessage
+_ba.chatmessage = new_chat_message
+
```

Comment:
we'll also need to add `_ba.set_party_icon_always_visible(True)` so players are able to open chat window to send the chat command while in co-op mode (the icon seems to be hidden by default)